corehq/util/mixin.py

```diff
@@ -29,3 +29,10 @@ def __init__(self, *args, **kwargs):
             value = getattr(self, field_name)
             if not value:
                 setattr(self, field_name, uuid.uuid4().hex)
+
+
+# https://gist.github.com/glarrain/5448253
+class ValidateModelMixin(object):
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        super(ValidateModelMixin, self).save(*args, **kwargs)
```

Comment:
This seems small enough that it'd be simpler in the model itself, rather than as a mixin.  This way it's harder to tell from reading the code that `full_clean` gets called on save.  That's a preference thing, so feel free to disagree.